rain_api_core/auth.py

```diff
@@ -0,0 +1,154 @@
+import contextlib
+import dataclasses
+import json
+import urllib.request
+from http.cookies import CookieError, SimpleCookie
+from time import time
+from typing import List, Mapping, Union
+from wsgiref.handlers import format_date_time as format_7231_date
+
+import jwt
+
+from rain_api_core.logging import get_log
+
+log = get_log()
+
+
+@dataclasses.dataclass
+class UserProfile:
+    user_id: str
+    token: str
+    groups: List[str]
+    first_name: str
+    last_name: str
+    email: str
+    iat: int = None
+    exp: int = None
+
+    @staticmethod
+    def from_jwt_payload(payload):
+        return UserProfile(
+            user_id=payload.get('urs-user-id'),
+            token=payload.get('urs-access-token'),
+            groups=payload.get('urs-groups'),
+            first_name=payload.get('first_name'),
+            last_name=payload.get('last_name'),
+            email=payload.get('email'),
+            iat=payload.get('iat'),
+            exp=payload.get('exp')
+        )
+
```

Comment:
I'd probably go with a `classmethod` and use the `cls` parameter here over a `staticmethod`